packages/erk-shared/tests/unit/prompt_executor/test_real.py

```diff
@@ -0,0 +1,196 @@
+"""Tests for RealPromptExecutor retry behavior.
```

Comment:
tests in tests/unit should use fakes not mocks  
  
try to build some tests with fakes  
  
move mock-based tests to integration test folder

Reply:
Resolved via /erk:pr-address at 2025-12-31 05:42 AM PST - Moved mock-based tests to packages/erk-shared/tests/integration/prompt_executor/test_real.py. Created fake-based unit tests in packages/erk-shared/tests/unit/prompt_executor/test_real.py that test retry constants and FakePromptExecutor transient_failures behavior.

_Addressed via `/erk:pr-address` at 2025-12-31 05:42 PST_